libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -1453,6 +1454,7 @@ void Compass::_detect_backends(void)
                         continue;
                     }
                     _priority_did_stored_list[i].set_and_save(detected_devid);
+                    _priority_did_stored_list[i].set_default(detected_devid);
```

Comment:
Looks like this code path only happens if something has gone missing. There is no point in setting the default in that case because it will only be set one time not on every boot.

Reply:
it is useful to set it as it means we don't show it unless the user has changed it. This value is being set by ArduPilot, not being set by the user. Try it, you'll find it works

Reply:
I'm sure it does work. My point is that this code path only happens if something has gone wrong? So we only show the correct default after something has gone wrong but not in normal operation?

>it is useful to set it as it means we don't show it unless the user has changed it.

This is true, but it only does that on this boot. The next boot it will still have been changed by AP and not the user but we don't show that with the defaults.

Maybe I just cant find where we set the default normally, this compass priority stuff is complicated.

Reply:
It looks like there in your none defaults list:

```
COMPASS_PRIO1_ID       0.0000   97539.0000
COMPASS_PRIO2_ID       0.0000  131874.0000
COMPASS_PRIO3_ID       0.0000  263178.0000
```

So this change does not work in normal operation?